### PR TITLE
[DeadCode] Skip RemoveUnusedVariableAssignRector if variable used before or in assign expr

### PR DESCRIPTION
**Pull request by Androbin**
Closes #173
Closes #205 @samsonasik 
Fixes rectorphp/rector#6494

### CODE REVIEW

**Comment by samsonasik** [comment]
Thank you, please also provide the fix

**Comment by Androbin** [comment]
@samsonasik The PR is now ready for review

**Comment by Androbin** [comment]
@samsonasik @TomasVotruba The PR is now ready for review

File: src/PhpParser/Comparing/ConditionSearcher.php [hunk 1]
```diff
@@ -4,72 +4,102 @@
...
-                continue;
-            }
+        if (!$this->searchForVariableRedeclaration($varNode, $if->stmts)) {
+            return false;
+        }
 
-            if (! $statementIf->expr instanceof Assign) {
-                continue;
-            }
+        if (!$this->searchForVariableRedeclaration($varNode, $elseNode->stmts)) {
```
Comment (Androbin) [inline src/PhpParser/Comparing/ConditionSearcher.php:39]: We could easily extend this to cover `elif` nodes

**Comment by samsonasik** [comment]
Thank you, please run ecs fix:

```bash
vendor/bin/ecs check --fix src/PhpParser/Comparing/ConditionSearcher.php

**Review by samsonasik — APPROVED** [review]

**Comment by TomasVotruba** [comment]
Thank you @Androbin :clap:

**Pull request merged**